quartz-core/src/main/java/org/quartz/impl/StdSchedulerFactory.java

```diff
@@ -492,7 +494,16 @@ private Properties overrideWithSysProps(Properties props) {
         }
 
         if (sysProps != null) {
-            props.putAll(sysProps);
+            // Use the propertyNames to iterate to avoid 
+            // a possible ConcurrentModificationException
+            Enumeration<?> en = sysProps.propertyNames();
```

Comment:
This is going to throw `ClassCastException` if any of the keys aren't strings, which is unfortunately possible here. We could instead use `stringPropertyNames()` which omits the non-String keys. Or, horror of horrors we could instead call `props.putAll((Properties) sysProps.clone())`

Reply:
Quite right, thanks. I've fixed with the latest commit and didn't use `stringPropertyNames` because there is still a race condition where a value could be changed to an Object just after I've read `stringPropertyNames`.